core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -561,7 +561,12 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
       val consumerRebalanceListener = new InternalRebalanceListenerForNewConsumer(this, customRebalanceListener)
       if (whitelistOpt.isDefined) {
         try {
-          consumer.subscribe(Pattern.compile(whitelistOpt.get), consumerRebalanceListener)
+          val revisedWhitelist = whitelistOpt.get.trim
+            .replace(',', '|')
+            .replace(" ", "")
+            .replaceAll("""^["']+""","")
+            .replaceAll("""["']+$""","") // property files may bring quotes
+          consumer.subscribe(Pattern.compile(revisedWhitelist), consumerRebalanceListener)
```

Comment:
Looks like you committed more than needed in the last try?
On the unit test note, I'm wondering if it makes sense to add (at least) one unit test that verifies the issue raised in the JIRA would not occur with this patch?